tests/python/gaia-ui-tests/gaiatest/tests/functional/lockscreen/test_lockscreen_time_check.py

```diff
@@ -0,0 +1,75 @@
+from marionette.wait import Wait
+
+from gaiatest.apps.settings.app import Settings
+from gaiatest import GaiaTestCase
+from gaiatest.apps.lockscreen.app import LockScreen
+
+
+class TestLockScreen(GaiaTestCase):
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        self.connect_to_local_area_network()
+
+    def test_lockscreen_time_check(self):
+
+        self.settings = Settings(self.marionette)
+        self.settings.launch()
+        datetime = self.settings.open_date_and_time_settings()
+        datetime.set_automatic_time_update(False)
+
+        # record time and change the region
+        old_time = datetime.get_current_time_text
+        datetime.set_region('Atlantic Ocean')
+
+        # get the displayed time after the region change
+        new_time = datetime.get_current_time_text
+
+        # lock screen and check time on the lockscreen
+        self.marionette.switch_to_frame()
+        self.device.lock()
+        lock_screen = LockScreen(self.marionette)
+        self.assertTrue(self.get_time_difference(new_time, lock_screen.time) < 5)
+
+        # configure to set the time automatically (this will revert the timezone change), then lock screen
+        lock_screen.switch_to_frame()
+        lock_screen.unlock()
+        self.marionette.switch_to_frame(self.apps.displayed_app.frame)
+        datetime.set_automatic_time_update(True)
+        self.marionette.switch_to_frame()
+        self.device.lock()
+
+        # wait until device is off and turn back on to check that the time is changed
+        Wait(self.marionette, timeout=20).until(
+            lambda m: not self.device.is_screen_enabled)
+        self.device.turn_screen_on()
+
+        # Check it reverted to the correct time.
+        # Allow 10 minutes difference max (since treeherder tests take longer)
+        self.assertTrue(self.get_time_difference(old_time, lock_screen.time) < 10)
+
+
+    def get_time_difference(self, time_a, lockscreen_time):
```

Comment:
It's slightly different, because in the music app we're getting the elapsed time (starts from 0:00), but in this case I need the difference between two times, and having AM/PM complicated things slightly.  for example, 12:59 to 1:02 is 3 minutes, but you have to make sure 12 = 0